lib/rubocop/cop/style/character_literal.rb

```diff
@@ -32,13 +32,11 @@ def autocorrect(node)
 
         # Dummy implementation of method in ConfigurableEnforcedStyle that is
         # called from StringHelp.
-        def opposite_style_detected
-        end
+        def opposite_style_detected; end
```

Comment:
Hm. I think you might be confusing the first `;` in the definition and the (optional) second one when there's a method body, i.e.:

```
def foo; bar end
```

Defining without the `;` will raise:

```
SyntaxError: (irb):1: syntax error, unexpected keyword_end, expecting ';' or '\n'
```

😊